tests/by-util/test_mktemp.rs

```diff
@@ -571,6 +572,12 @@ fn test_template_path_separator() {
         ));
 }
 
+/// Test that a prefix with a point is valid.
+#[test]
+fn test_prefix_template_separator() {
+    new_ucmd!().args(&["-t", TEST_TEMPLATE9]).succeeds();
```

Comment:
This does not really test the issue right? I thought it needed `-p` to error on `main` or am I mistaken?

Reply:
I believe it is a mistake, I checked the code and the flow that leads to the error is still correct. If ```-p``` is not passed as a parameter it will use a default OS temp directory, does not affect the problem.

Only the ```-t``` parameter is needed for the test as I understand it.

Reply:
I tested it by changing the TEST_TEMPLATE9 with ```/``` instead of ```.```, it seems to be correct. Crashes and fail when I use  ```/```


```sh
running 1 test
test test_mktemp::test_prefix_template_separator ... FAILED

failures:

---- test_mktemp::test_prefix_template_separator stdout ----
run: /home/zau/Desktop/Projects/CONTRIB/coreutils/target/debug/coreutils mktemp -t a/XXXX
thread 'test_mktemp::test_prefix_template_separator' panicked at 'Command was expected to succeed.
stdout = 
 stderr = mktemp: invalid template, 'a/XXXX', contains directory separator
', tests/by-util/test_mktemp.rs:578:47
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Interesting! Let's add both cases to the test suite. It never hurts to have more tests :)

Reply:
Done, it was fun

Reply:
Wait, I still don't get it. The case you've written succeeds on `main`. We do still need to verify that the fix fixed the issue, right?

For example something like this:
```rust
let (at, mut ucmd) = at_and_ucmd!();
at.mkdir("some");
at.mkdir("some/dir");
ucmd.args(&["-p", "some/dir", "-t", TEST_TEMPLATE9]).succeeds();
```

Reply:
It's really strange that ```-p``` is needed for the test, I checked as I said and indeed the test passes the main. I think we can simplify it to:

```rust
new_ucmd!().args(&["-p", ".", "-t", TEST_TEMPLATE9]).succeeds();
```

Reply:
I'll upload the test correction and check the why of this `-p`